### PR TITLE
Add silent command

### PR DESCRIPTION
**Pull request by 0xdeaddood**
This PR adds the `silent command` option in wmiexec.py, atexec.py and dcomexec.py. It is based on this PR #605.

Additionally, it fixes an error in atexec.py when the script is executed without the silent command option.

### CODE REVIEW

**Comment by martingalloar** [comment]
LGTM! 🚀 

As a small piece of feedback, maybe we can clarify on the help description that this is silent in the sense that it runs without calling `cmd.exe` but also that there's no output 😄 Otherwise we might have users expecting the command output and thinking it's not working.

File: examples/atexec.py [hunk 1]
```diff
@@ -235,8 +241,8 @@ def cmd_split(cmdline):
     parser.add_argument('target', action='store', help='[[domain/]username[:password]@]<targetName or address>')
     parser.add_argument('command', action='store', nargs='*', default=' ', help='command to execute at the target ')
     parser.add_argument('-session-id', action='store', type=int, help='an existed logon session to use (no output, no cmd.exe)')
-
     parser.add_argument('-ts', action='store_true', help='adds timestamp to every logging output')
+    parser.add_argument('-silentcommand', action='store_true', default = False, help='does not execute cmd.exe to run given command ')
```
Comment (martingalloar) [inline examples/atexec.py:245]: We might add here that there's no output generated

File: examples/dcomexec.py [hunk 2]
```diff
@@ -554,7 +567,8 @@ def load_smbclient_auth_file(path):
 
     parser.add_argument('command', nargs='*', default = ' ', help='command to execute at the target. If empty it will '
                                                                   'launch a semi-interactive shell')
-
+    parser.add_argument('-silentcommand', action='store_true', default = False, help='does not execute a command '
+                        'proccessor to run given command (cannot run dir/cd/etc.)')
```
Comment (martingalloar) [inline examples/dcomexec.py:571]: We might add here that there's no output generated. Also we might want to unify help messages.

File: examples/wmiexec.py [hunk 3]
```diff
@@ -295,97 +300,101 @@ def send_data(self, data):
...
-                                                                           '(default ADMIN$)')
-    parser.add_argument('-nooutput', action='store_true', default = False, help='whether or not to print the output '
-                                                                                '(no SMB connection created)')
+    parser.add_argument('-share', action='store', default='ADMIN$', help='share where the output will be grabbed from '
+                                                                         '(default ADMIN$)')
+    parser.add_argument('-nooutput', action='store_true', default=False, help='whether or not to print the output '
+                                                                              '(no SMB connection created)')
     parser.add_argument('-ts', action='store_true', help='Adds timestamp to every logging output')
+    parser.add_argument('-silentcommand', action='store_true', default=False,
+                        help='does not execute cmd.exe to run given command ')
```
Comment (martingalloar) [inline examples/wmiexec.py:366]: Same here, we might want to add that there's no output generated.

**Comment by martingalloar** [comment]
Let us know what you think about this changes and the new PR @aconite33!

**Comment by aconite33** [comment]
> Let us know what you think about this changes and the new PR @aconite33!

oh snap! Let me take a look!

**Comment by aconite33** [comment]
Got wmiexec.py working. I couldn't get Dcomexec.py working, keept getting errors:


`[-] DCOM SessionError: unknown error code: 0x800706ba`

I've ran into that before with DCOM, I'm forgetting how I confirmed I've gotten execution before with it. Tried using both -silentcommand and leaving it off, along with the different -object methods.

**Comment by 0xdeaddood** [comment]
> Got wmiexec.py working. I couldn't get Dcomexec.py working, keept getting errors:
> 
> `[-] DCOM SessionError: unknown error code: 0x800706ba`
> 
> I've ran into that before with DCOM, I'm forgetting how I confirmed I've gotten execution before with it. Tried using both -silentcommand and leaving it off, along with the different -object methods.

Have you checked the firewall configuration?

**Comment by aconite33** [comment]
> Have you checked the firewall configuration?

Well, now I feel like an idiot. Yep! Both work! Rock on!

I feel good about this one! Thanks for taking the time to look at this one! Can't wait to get this in the main branch :)

**Comment by 0xdeaddood** [comment]
Thanks for the confirmation @aconite33! It's a great addition! Merging...

**Pull request merged**